server/memphis_handlers_functions.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2022-2023 The Memphis.dev Authors
+// Licensed under the Memphis Business Source License 1.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// Changed License: [Apache License, Version 2.0 (https://www.apache.org/licenses/LICENSE-2.0), as published by the Apache Foundation.
+//
+// https://github.com/memphisdev/memphis/blob/master/LICENSE
+//
+// Additional Use Grant: You may make use of the Licensed Work (i) only as part of your own product or service, provided it is not a message broker or a message queue product or service; and (ii) provided that you do not use, provide, distribute, or make available the Licensed Work as a Service.
+// A "Service" is a commercial offering, product, hosted, or managed service, that allows third parties (other than your own employees and contractors acting on your behalf) to access and/or use the Licensed Work or a substantial set of the features or functionality of the Licensed Work to third parties as a software-as-a-service, platform-as-a-service, infrastructure-as-a-service or other similar services that compete with Licensor products or services.
+package server
+
+import (
+	"time"
+
+	"fmt"
+	"memphis/models"
+
+	"github.com/gin-gonic/gin"
+)
+
+type ContentYamlFile struct {
+	FunctionName string   `json:"function_name" yaml:"function_name"`
+	Description  string   `json:"description" yaml:"description"`
+	Tags         []string `json:"tags" yaml:"tags"`
+	Language     string   `json:"language" yaml:"language"`
+}
+
+type FunctionsResult struct {
+	FunctionName string    `json:"function_name"`
+	Description  string    `json:"description"`
+	Tags         []string  `json:"tags"`
+	Language     string    `json:"language"`
+	LastCommit   time.Time `json:"last_commit"`
+	Link         string    `json:"link"`
+	Repository   string    `json:"repository"`
+	Branch       string    `json:"branch"`
+}
+
+type FunctionsHandler struct{}
+
+func (fh FunctionsHandler) GetAllFunctions(c *gin.Context) {
+	user, err := getUserDetailsFromMiddleware(c)
+	if err != nil {
+		serv.Errorf("GetAllFunctions at getUserDetailsFromMiddleware: %v", err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+
+	connectedRepos, err := fh.GetConnectedSourceCodeRepos(user.TenantName)
```

Comment:
Again, please write the things we are talking about it's the third time we are talking on the same exactly thing.
This function should be part of the GetContentOfSelectedRepos